### ISSUE TITLE
`get` and `find` should have a `--yaml` flag and a `--human` flag

### ISSUE THREAD

**Issue by evanfarrar** [cloudfoundry-incubator/credhub-cli#24]
Currently these commands output YAML by default. However, conceivably, this CLI was meant to be used by humans who aren't usually interested in `version_created_at` or the database ID. They mainly want to know the value of their password or the name of their password, not the metadata associated with it. 

Currently the `find` command returns this YAML:
```
jumpbox/0:~$ ./credhub find -p /bosh-ci-evanfarrar/concourse
credentials:
- name: /bosh-ci-evanfarrar/concourse/atc
  version_created_at: 2017-11-09T21:11:09Z
- name: /bosh-ci-evanfarrar/concourse/atc_ca
  version_created_at: 2017-11-09T18:48:03Z
- name: /bosh-ci-evanfarrar/concourse/database_password
  version_created_at: 2017-11-09T18:48:02Z
- name: /bosh-ci-evanfarrar/concourse/database_username
  version_created_at: 2017-11-09T18:48:02Z
- name: /bosh-ci-evanfarrar/concourse/database_name
  version_created_at: 2017-11-09T18:48:02Z
- name: /bosh-ci-evanfarrar/concourse/basic_auth_password
  version_created_at: 2017-11-09T18:48:02Z
- name: /bosh-ci-evanfarrar/concourse/basic_auth_username
  version_created_at: 2017-11-09T18:48:02Z
```

And the `get` command returns this YAML:
```
jumpbox/0:~$ ./credhub get --name `./credhub find -n "basic_auth_username" | grep -e"- name" | cut -d':' -f2`
id: e6f1d764-37a1-4261-86fd-64fb4ce6e02d
name: /bosh-ci-evanfarrar/concourse/basic_auth_username
type: password
value: APASSWORD
version_created_at: 2017-11-09T18:48:02Z
```

I propose that we make the output human-readable and human-relevant by default, but provide the YAML format as an option. It is arguably a smaller burden for a computer to remember and type an extra flag than it is for a human to do so.

new proposed find output:
```
jumpbox/0:~$ ./credhub find -p /bosh-ci-evanfarrar/concourse
/bosh-ci-evanfarrar/concourse/atc
/bosh-ci-evanfarrar/concourse/atc_ca
/bosh-ci-evanfarrar/concourse/database_password
/bosh-ci-evanfarrar/concourse/database_username
/bosh-ci-evanfarrar/concourse/database_name
/bosh-ci-evanfarrar/concourse/basic_auth_password
/bosh-ci-evanfarrar/concourse/basic_auth_username
```

new proposed get output:
```
jumpbox/0:~$ ./credhub get --name `./credhub find -n "basic_auth_username"`
APASSWORD
```

A possible migration path:
* 1.X: introduce `--human` to output just the value or just the keys delimited by newlines, introduce `--yaml` flag which does nothing as it is already the default.
* 2.X: `--human` now does nothing but print a deprecation warning, as it is the default. `--yaml` does what the default used to do.
* 3.x: remove `--human`

Related: this is an alternative suggestion for #3

**Comment by crawsible** [comment]
An interesting idea. I agree that YAML isn't a great default. I'd like to understand more of the operations you and BBL users are trying to do, like logging into a cf you just deployed or trying to pre-seed a value so it gets used when the deploy occurs, rotating values, etc. Could you tell me more about what you've been using CredHub to do/script in the last few weeks and months?

**Comment by evanfarrar** [comment]
I bumped up against this when documenting the install process for bbl+my concourse-deployment manifest. Particular. When you are using a file based secret store you kind of rely on the fact that 

I documented my experience here:
https://github.com/cloudfoundry/bosh-bootloader/blob/master/docs/concourse-gcp.md

I don't _necessarily_ think this has to be in the credhub CLI: If I could use BOSH interpolate commands to fetch data out of the deployment, I'd be open to that. The main things that attract me to this suggestion are 1) an interface that is easy for me to discover the name of the field I'm looking for when I might not exactly know it by heart (I just knew it had "admin" and "password" so `credhub -f | grep` would help)  2) output that is pipe-able to make one-liners possible for my documentation
